<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QStackedWidget" name="pages">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1200</width>
      <height>700</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="main_page">
     <widget class="QWidget" name="horizontalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>1191</width>
        <height>702</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="main_page_layout" stretch="2,4">
       <property name="spacing">
        <number>3</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>50</height>
            </size>
           </property>
           <property name="title">
            <string>цвет фона</string>
           </property>
           <widget class="QWidget" name="horizontalLayoutWidget_5">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>381</width>
              <height>32</height>
             </rect>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QWidget" name="bg_color_preview" native="true">
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>16777215</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_bg_color_change">
               <property name="text">
                <string>Поменять</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>50</height>
            </size>
           </property>
           <property name="title">
            <string>цвет линии</string>
           </property>
           <widget class="QWidget" name="horizontalLayoutWidget_6">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>381</width>
              <height>32</height>
             </rect>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QWidget" name="line_color_preview" native="true">
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>16777215</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_line_color_change">
               <property name="text">
                <string>Поменять</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_7">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>50</height>
            </size>
           </property>
           <property name="title">
            <string>Алгоритм</string>
           </property>
           <widget class="QComboBox" name="algorithm_selection">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>379</width>
              <height>32</height>
             </rect>
            </property>
            <item>
             <property name="text">
              <string>Канонического уравнения</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Параметрического уравнения</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Брезенхема</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Средней точки</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Библиотечный</string>
             </property>
            </item>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_3">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>180</height>
            </size>
           </property>
           <property name="title">
            <string>Построение окружности/эллипса</string>
           </property>
           <widget class="QWidget" name="verticalLayoutWidget">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>381</width>
              <height>165</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,0,2,1">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <item>
                <widget class="QLabel" name="label">
                 <property name="maximumSize">
                  <size>
                   <width>160</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Координаты центра (x:y)</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="center_x">
                 <property name="text">
                  <string>399</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="center_y">
                 <property name="text">
                  <string>349</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_11">
               <item>
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>Окружность</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>Эллипс</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_5">
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_7">
                   <item>
                    <widget class="QLabel" name="label_4">
                     <property name="text">
                      <string>Радиус (R)</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="radius"/>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_4">
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_9">
                   <item>
                    <widget class="QLabel" name="label_5">
                     <property name="text">
                      <string>Ширина (Rx)</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="radius_x"/>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_10">
                   <item>
                    <widget class="QLabel" name="label_6">
                     <property name="text">
                      <string>Высота (Ry) </string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="radius_y"/>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <widget class="QPushButton" name="btn_circle_draw">
                 <property name="text">
                  <string>Построить окружность</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="btn_ellipse_draw">
                 <property name="text">
                  <string>Построить эллипс</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_6">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>230</height>
            </size>
           </property>
           <property name="title">
            <string>Построение спектра окружности/эллипса</string>
           </property>
           <widget class="QWidget" name="verticalLayoutWidget_6">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>381</width>
              <height>211</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_6" stretch="1,2,1,1,1,1">
             <property name="spacing">
              <number>1</number>
             </property>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_15">
               <item>
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>Окружность</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>Эллипс</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_14">
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_16">
                 <item>
                  <widget class="QLabel" name="label_11">
                   <property name="text">
                    <string>Нач. радиус</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="start_radius"/>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_7">
                 <property name="spacing">
                  <number>0</number>
                 </property>
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_17">
                   <item>
                    <widget class="QLabel" name="label_12">
                     <property name="text">
                      <string>Нач. ширина</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="start_radius_x"/>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_18">
                   <item>
                    <widget class="QLabel" name="label_13">
                     <property name="text">
                      <string>Нач. высота</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="start_radius_y"/>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_13">
               <item>
                <widget class="QLabel" name="label_9">
                 <property name="minimumSize">
                  <size>
                   <width>200</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Шаг изменения</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="step"/>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_20">
               <item>
                <widget class="QLabel" name="label_10">
                 <property name="minimumSize">
                  <size>
                   <width>200</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Количество фигур</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="figures_count"/>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_19">
               <item>
                <widget class="QPushButton" name="btn_draw_circles_spectrum">
                 <property name="text">
                  <string>Постр спектр окр-ей</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="btn_draw_ellipses_spectrum">
                 <property name="text">
                  <string>Постр спектр элл-ов</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_12">
               <item>
                <widget class="QPushButton" name="btn_time_cmp_circles">
                 <property name="text">
                  <string>Время пост-ия окр-ей</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="btn_time_cmp_ellipses">
                 <property name="acceptDrops">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>Время пост-ия эл-ов</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_4">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>105</height>
            </size>
           </property>
           <property name="title">
            <string>Управление</string>
           </property>
           <widget class="QWidget" name="verticalLayoutWidget_3">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>381</width>
              <height>91</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <property name="spacing">
              <number>1</number>
             </property>
             <item>
              <widget class="QGroupBox" name="groupBox_5">
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>60</height>
                </size>
               </property>
               <property name="title">
                <string>Масштаб</string>
               </property>
               <widget class="QWidget" name="horizontalLayoutWidget_2">
                <property name="geometry">
                 <rect>
                  <x>0</x>
                  <y>20</y>
                  <width>351</width>
                  <height>41</height>
                 </rect>
                </property>
                <layout class="QHBoxLayout" name="horizontalLayout_2">
                 <item>
                  <widget class="QSlider" name="scale_slider">
                   <property name="minimum">
                    <number>100</number>
                   </property>
                   <property name="maximum">
                    <number>1500</number>
                   </property>
                   <property name="singleStep">
                    <number>50</number>
                   </property>
                   <property name="pageStep">
                    <number>100</number>
                   </property>
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="tickPosition">
                    <enum>QSlider::TicksAbove</enum>
                   </property>
                   <property name="tickInterval">
                    <number>100</number>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="slider_value">
                   <property name="minimumSize">
                    <size>
                     <width>50</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="font">
                    <font>
                     <pointsize>15</pointsize>
                    </font>
                   </property>
                   <property name="text">
                    <string>100%</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_clear_screen">
               <property name="text">
                <string>Очистить экран</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGraphicsView" name="graphicsView">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>800</width>
           <height>700</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>800</width>
           <height>700</height>
          </size>
         </property>
         <property name="backgroundBrush">
          <brush brushstyle="NoBrush">
           <color alpha="255">
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="time_cmp_page">
     <widget class="QPushButton" name="back_btn">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>100</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Назад</string>
      </property>
     </widget>
     <widget class="QWidget" name="verticalLayoutWidget_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>1181</width>
        <height>661</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="time_layout"/>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
